src/analysis.zig

```diff
@@ -3236,7 +3259,18 @@ fn iterateSymbolsContainerInternal(
                 .simple_var_decl,
                 .aligned_var_decl,
                 => {
-                    if (instance_access) continue;
+                    if (instance_access) {
+                        // allow declarations which evaluate to functions where
+                        // the first parameter has the type of the container:
+                        const alias = try analyser.resolveVarDeclAlias(
+                            NodeWithHandle{ .node = node, .handle = handle },
+                        ) orelse continue;
+
+                        const alias_type = try alias.resolveType(analyser) orelse continue;
+
+                        const container_type = TypeWithHandle.typeVal(container_handle);
+                        if (!try analyser.isSelfFunction(alias_type, container_type)) continue;
```

Comment:
I didn't notice this previously, but `resolveVarDeclAlias` will only resolve things like imports, symbol lookups and field accesses but the expression could be more complex than that which can be handled by `resolveTypeOfNode`.
```suggestion
                        const alias = try analyser.resolveTypeOfNode(
                            NodeWithHandle{ .node = node, .handle = handle },
                        ) orelse continue;

                        const container_type = TypeWithHandle.typeVal(container_handle);
                        if (!try analyser.isSelfFunction(alias, container_type)) continue;
```